functions/src/index.ts

```diff
@@ -86,3 +111,94 @@ export const deleteIndexOnThoughtIndexDelete = functions.database
       console.warn(err, 'index delete error')
     }
   })
+
+// Sends feedback recieved from authenticated or anonymous user to the em support.
+export const sendFeedbackEmail = functions.https.onRequest(async (request, response) =>
+  // TODO: Allow requests from specific origin in production
+  cors({ origin: true })(request, response, async () => {
+
+    const { userId, feedback } = request.body
+
+    if (!feedback) {
+      response.status(400).json({
+        message: 'Feedback is required.'
+      })
+      return
+    }
+
+    const ip = request.headers['x-appengine-user-ip'] || request.headers['x-forwarded-for'] || request.connection.remoteAddress
+
+    // rate limit by ip address
+    try {
+      await rateLimiter.rejectOnQuotaExceededOrRecordUsage(encode(ip as string))
+    }
+    catch (err) {
+      response.status(429).json({
+        message: 'Your feedback submission has been rate limited.'
+      })
+      return
+    }
+
+    const getUserNameAndEmail = async () => {
+      const userRef = database.ref(`users/${userId}`)
+      const nameRef = userRef.child('name')
+      const emailRef = userRef.child('email')
+
+      const [nameSnapshot, emailSnapshot] = await Promise.all([nameRef.once('value'), emailRef.once('value')])
+
+      return {
+        name: nameSnapshot.val() as string,
+        email: emailSnapshot.val() as string
+      }
+    }
+
+    const feedbackTemplate = readFileSync(path.resolve(__dirname, 'templates', 'feedback.hbs')).toString('utf8')
+    const template = handlebar.compile(feedbackTemplate)
+    const userDetail = userId ? await getUserNameAndEmail() : null
+
+    if (userId && userDetail && !userDetail.name) {
+      response.status(400).json({
+        message: 'User id is invalid.'
+      })
+      return
+    }
+
+    // Fix: type not working for FormData
+    const mailgun = new Mailgun(formData as any)
+
+    // mailgun configs
+    const MAILGUN_API_KEY = functions.config().mailgun.api_key
+    const MAILGUN_DOMAIN = functions.config().mailgun.domain
+    const MAILGUN_FEEDBACK_EMAIL = functions.config().mailgun.feedback_email
+    const MAILGUN_SUPPORT_EMAIL = functions.config().mailgun.em_support_email
+
+    const mailgunClient = mailgun.client({
+      username: 'api',
+      key: MAILGUN_API_KEY,
+    })
+
+    try {
+      await mailgunClient.messages.create(
+        MAILGUN_DOMAIN,
+        {
+          from:
+            `Em Feedback <${userDetail?.email ?? MAILGUN_FEEDBACK_EMAIL}>`,
+          to: [MAILGUN_SUPPORT_EMAIL],
+          subject: 'App Feedback',
+          html: template({
+            feedback,
+            name: userDetail?.name ?? 'Anonymous User',
+            email: userDetail?.email
+          }),
+        }
+      )
+    }
+    catch (err) {
+      console.error(err, 'Mailgun error')
+    }
```

Comment:
You'll want to send an HTTP error response here, otherwise it will send a success to the client.

Reply:
Ahh I missed that. Will fix this.